Python/Tests/Core.UI/EnvironmentUITests.cs

```diff
@@ -507,5 +505,82 @@ TreeNode env
                 EnvironmentReplWorkingDirectoryTest(app, project, env);
             }
         }
+
+        public void SwitcherSingleProject(PythonVisualStudioApp app) {
+            var globalDefault37 = PythonPaths.Python37_x64 ?? PythonPaths.Python37;
+            globalDefault37.AssertInstalled();
+
+            var added27 = PythonPaths.Python27_x64 ?? PythonPaths.Python27;
+            added27.AssertInstalled();
+
+            var added36 = PythonPaths.Python36_x64 ?? PythonPaths.Python36;
+            added36.AssertInstalled();
+
+            using (var dis = app.SelectDefaultInterpreter(globalDefault37)) {
+                // Project has no references, uses global default
+                var project = CreateTemporaryProject(app);
+                CheckSwitcherText(app, globalDefault37.Configuration.Description);
+
+                // Project has one referenced interpreter
+                app.ServiceProvider.GetUIThread().Invoke(() => {
+                    var pp = project.GetPythonProject();
+                    pp.AddInterpreter(added27.Configuration.Id);
+                });
+                CheckSwitcherText(app, added27.Configuration.Description);
+
+                // Project has two referenced interpreters (active remains the same)
+                app.ServiceProvider.GetUIThread().Invoke(() => {
+                    var pp = project.GetPythonProject();
+                    pp.AddInterpreter(added36.Configuration.Id);
+                });
+                CheckSwitcherText(app, added27.Configuration.Description);
+
+                // No switcher visible when solution closed and no file opened
+                app.Dte.Solution.Close(SaveFirst: false);
+                CheckSwitcherHidden(app);
+            }
+        }
+
+        public void SwitcherNoProject(PythonVisualStudioApp app) {
+            var globalDefault37 = PythonPaths.Python37_x64 ?? PythonPaths.Python37;
+            globalDefault37.AssertInstalled();
+
+            using (var dis = app.SelectDefaultInterpreter(globalDefault37)) {
+                // Loose Python file shows global default
+                app.OpenDocument(TestData.GetPath("TestData", "Environments", "Program.py"));
+                CheckSwitcherText(app, globalDefault37.Configuration.Description);
+
+                // No switcher visible when solution closed and no file opened
+                app.Dte.ActiveWindow.Close();
```

Comment:
I remember these things like `ActiveWindow` can easily be nulls in DTE.

Reply:
The test just opened a document right before, so if active window is null I'm ok with test failing.